app/lib/actions/candlepin/consumer/destroy.rb

```diff
@@ -9,6 +9,10 @@ class Destroy < Candlepin::Abstract
         def run
           ::Katello::Resources::Candlepin::Consumer.destroy(input[:uuid])
         end
+
+        def finalize
+          ::Katello::Pool.import_all
+        end
```

Comment:
This really belongs in a different action, such as  katello/system/destroy.rb

also, reindexing all pools across all orgs probably isn't ideal.  We should try to limit this to a single org, or potentially just the pools the system was consuming?